doc/doxygen/CMakeLists.txt

```diff
@@ -81,6 +81,28 @@ ELSE()
 ENDIF()
 
 
+# Collect where the build system will write the .inst files after
+# expansion from the .inst.in files. We need this because we let
+# doxygen read the .cc files and they #include the .inst files.
+#
+# To get at this, find all of the .inst.in files, strip all but the
+# directory name under $CMAKE_SOURCE_DIR/source, and then prepend
+# the build directory to it
+FILE(GLOB _inst_in_files
+     "${CMAKE_SOURCE_DIR}/source/*/*.inst.in")
```

Comment:
My only concern is that CMake won't update requirements if things are cached, but these change so infrequently (and the change will only effect doxygen errors) that I am not worried about it.

Reply:
Yes, I agree. I don't know what to do about this issue -- cmake doesn't seem to have a good way to do this, but it's also of minor concern because the worst that can happen is that doxygen complains about a missing file it wants to read but from which it nevertheless doesn't create any output.

It's also really just new directories that cause the issue. We only use the file list above to extract the source directories in which these files are located. So you'd have to add a whole new directory to trip up this logic. That does not happen very often.